nixos/modules/profiles/base.nix

```diff
@@ -51,7 +51,7 @@
   ];
 
   # Include support for various filesystems.
-  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "zfs" "ntfs" "cifs" ];
+  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "nilfs2" "xfs" "zfs" "ntfs" "cifs" ];
```

Comment:
I'm not sure we want to add this by default. I liked your blog post on this software but I think we should probably only add items here that are a bit more popular.

My memory on the filesystem code is a bit hazy... but would it create an infinite recursion here if we conditionally added this value depending on whether the user has added this file system to their system via `config.fileSystems`?

Reply:
I saw reiserfs so I thought it would be fine. I can try to add it conditionally and see if it works.

Reply:
I removed it from the list, it's not useful to boot on a nilfs root filesystem. It seems to be used to provide tools in the installer ISO and I don't think we need to have nilfs-utils by default.